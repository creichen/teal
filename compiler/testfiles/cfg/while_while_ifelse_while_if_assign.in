fun f(a, b) = {
  while b {
    if a {
      while a {
        if b > 0 {
          b := a;
	}
      }
    } else {
      print(0);
    }
  }
}

