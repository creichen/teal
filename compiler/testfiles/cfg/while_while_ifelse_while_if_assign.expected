Entry#0 @ 1 :
  -> Access#0
VarDecl#0 @ 1 :
VarDecl#1 @ 1 :
Access#0 @ 2 :
  -> Access#1
  -> Exit#0
Access#1 @ 3 :
  -> Access#2
  -> IntConstant#1
Access#2 @ 4 :
  -> Access#0
  -> Access#3
Access#3 @ 5 :
  -> IntConstant#0
GTExpr#0 @ 5 :
  -> Access#2
  -> Access#5
IntConstant#0 @ 5 :
  -> GTExpr#0
Access#4 @ 6 :
  -> AssignStmt#0
AssignStmt#0 @ 6 :
  -> Access#2
Access#5 @ 6 :
  -> Access#4
CallExpr#0 @ 10 :
  -> Access#0
IntConstant#1 @ 10 :
  -> CallExpr#0
Exit#0 @ 13 :
